### PR TITLE
Update openspeedshop and cbtf-krell to use latest binutils version (fix #4992)

### LINKED ISSUE #4992: Issue while building openspeedshop  (older version of binutils?)
I am trying to build openspeedshop on Ubuntu 16.04 and I see following (while building cbtf-krell) : 

```
.............
  /home/guest/workarena/softwares/sources/spack/opt/spack/linux-ubuntu16-x86_64/gcc-5.4.1/binutils-2.24-dk6wlwesfajvuudbqzutq3fkx5kubp5t/bin/ld:
  final link failed: Bad value

  collect2: error: ld returned 1 exit status

  CMakeFiles/cmTC_91ae1.dir/build.make:97: recipe for target 'cmTC_91ae1'
  failed
  make[1]: *** [cmTC_91ae1] Error 1

  make[1]: Leaving directory
  '/tmp/guest/spack-stage/spack-stage-Wky2l2/cbtf-krell/build_cbtf_krell/CMakeFiles/CMakeTmp'
  Makefile:126: recipe for target 'cmTC_91ae1/fast' failed

  make: *** [cmTC_91ae1/fast] Error 2
  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:21 (project)

-- Configuring incomplete, errors occurred!

```

If I look into `CMakeFiles/CMakeError.log` I see:

```
/home/guest/workarena/softwares/sources/spack/lib/spack/env/gcc/gcc  -O2 -g     CMakeFiles/cmTC_91ae1.dir/testCCompiler.c.o  -o cmTC_91ae1 -rdynamic
/home/guest/workarena/softwares/sources/spack/opt/spack/linux-ubuntu16-x86_64/gcc-5.4.1/binutils-2.24-dk6wlwesfajvuudbqzutq3fkx5kubp5t/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/5/../../../x86_64-linux-gnu/crti.o: unrecognized relocation (0x2a) in section `.init'
/home/guest/workarena/softwares/sources/spack/opt/spack/linux-ubuntu16-x86_64/gcc-5.4.1/binutils-2.24-dk6wlwesfajvuudbqzutq3fkx5kubp5t/bin/ld: final link failed: Bad value
```

@jgalarowicz : Is this due to old version of binutils that openspeedshop depends on ?

```
depends_on("binutils@2.24+krellpatch", type='build')
```

### PR DESCRIPTION
**Pull request by pramodskumbhar**
tested on Ubuntu 16.04 with gcc-5.4.1.

### CODE REVIEW

File: var/spack/repos/builtin/packages/binutils/package.py [hunk 1]
```diff
@@ -39,10 +39,6 @@ class Binutils(AutotoolsPackage):
     version('2.23.2', '4f8fa651e35ef262edc01d60fb45702e')
     version('2.20.1', '2b9dc8f2b7dbd5ec5992c6e29de0b764')
 
-    # Add a patch that creates binutils libiberty_pic.a which is preferred by
-    # OpenSpeedShop and cbtf-krell
-    variant('krellpatch', default=False,
-            description="build with openspeedshop based patch.")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/binutils/package.py]: You can also remove the following line below:
```python
patch('binutilskrell-2.24.patch', when='@2.24+krellpatch')
```

File: var/spack/repos/builtin/packages/cbtf-krell/package.py [hunk 2]
```diff
@@ -75,7 +75,7 @@ class CbtfKrell(CMakePackage):
     depends_on("cmake@3.0.2:", type='build')
 
     # For binutils service
-    depends_on("binutils@2.24+krellpatch")
+    depends_on("binutils")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cbtf-krell/package.py]: @jgalarowicz Do these packages actually need the `binutils` dependency at all? See http://spack.readthedocs.io/en/latest/getting_started.html#binutils

File: var/spack/repos/builtin/packages/binutils/package.py (hunk 1 above)
Comment (pramodskumbhar) [inline var/spack/repos/builtin/packages/binutils/package.py]: done

File: var/spack/repos/builtin/packages/cbtf-krell/package.py (hunk 2 above)
Comment (jgalarowicz) [inline var/spack/repos/builtin/packages/cbtf-krell/package.py]: Yes, they both do need binutils.

**Comment by adamjstewart** [comment]
Don't worry about Codecov. We ignore it for package changes.

**Pull request merged**